### ISSUE TITLE
`assert` can only narrow `==`

### ISSUE THREAD

**Issue by lewis6991** [EmmyLuaLs/emmylua-analyzer-rust#734]
<img width="590" height="264" alt="Image" src="https://github.com/user-attachments/assets/e0bc03e8-1c89-4a94-8c14-369fba0e8866" />

```lua
local a --- @type string[]

assert(#a >= 1)

--- @type string
_ = a[1]

assert(#a == 1)

--- @type string
_ = a[1]

--- @type string
_2 = a[1]
```

There's also an unexpected diagnostic `Cannot assign string to integer`.

**Comment by CppCXY** [comment]
complete e5159a2441c9c15c97872e6eb62f9c46e35e1975
